rocAL_pybind/rocal_pybind.cpp

```diff
@@ -155,6 +160,110 @@ std::unordered_map<int, std::string> rocalToPybindOutputDtype = {
     {5, "int32"},
 };
 
+#if ENABLE_DLPACK
+    //dlpack functions
+    static bool supported_dl_device_type(const DLDeviceType &devType) {
+        switch (devType) {
+            case kDLROCM:
+            case kDLROCMHost:
+            case kDLCPU:
+                return true;
+            default:
+                return false;
+        }
+    }
+
+    static RocalTensorOutputType get_data_type(const DLDataType &dtype) {
+        if (dtype.lanes != 1) {
+            throw std::runtime_error("Data type lanes != 1 is not supported.");
+        }
+
+        switch (dtype.bits) {
+            case 8:
+                switch (dtype.code) {
+                    case kDLInt:
+                        return RocalTensorOutputType::ROCAL_INT8;
+                    case kDLUInt:
+                        return RocalTensorOutputType::ROCAL_UINT8;
+                    default:
+                        throw std::runtime_error(
+                            "Data type code is not supported.");
+                }
+            case 32:
+                switch (dtype.code) {
+                    case kDLInt:
+                        return RocalTensorOutputType::ROCAL_INT32;
+                    case kDLUInt:
+                        return RocalTensorOutputType::ROCAL_UINT32;
+                    case kDLFloat:
+                        return RocalTensorOutputType::ROCAL_FP32;
+                    default:
+                        throw std::runtime_error(
+                            "Data type code is not supported.");
+                }
+            case 16:
+            default:
+                throw std::runtime_error("Data type bits is not supported.");
+        }
+    }
+
+    static RocalOutputMemType get_data_location(const DLDeviceType &devType) {
+        switch (devType) {
+            case kDLROCM:
+                return RocalOutputMemType::ROCAL_MEMCPY_GPU;
+            case kDLROCMHost:
+            case kDLCPU:
+                return RocalOutputMemType::ROCAL_MEMCPY_HOST;
+            default:
+                throw std::runtime_error("Tensor device type is not supported.");
+        }
+    }
+
+    static DLDevice generate_dl_device(rocalTensor *rocal_tensor) {
+        DLDevice dev;
+
+        dev.device_id = 0;
```

Comment:
@LakshmiKumar23 In this function, the device ID is hardcoded as 0. Does this mean we are always creating DLTensors on device 0?